src/NuGet.Core/NuGet.Versioning/NuGetVersionFactory.cs

```diff
@@ -112,6 +104,169 @@ int IndexOf(string str, char c)
             }
         }
 
+        public static bool TryGetNormalizedVersion(string str, [NotNullWhen(true)] out Version? version)
```

Comment:
Make this `internal`. This is not an API I want external customers to have available. Our tests have `InternalsVisibleTo`.